drivers/acpi/scan.c

```diff
@@ -1382,6 +1382,13 @@ static void acpi_set_pnp_ids(acpi_handle handle, struct acpi_device_pnp *pnp,
 				for (i = 0; i < cid_list->count; i++)
 					acpi_add_id(pnp, cid_list->ids[i].string);
 			}
+		} else if (info->valid & ACPI_VALID_CID) {
+			cid_list = &info->compatible_id_list;
+			pr_info("%s: CID without HID for hardware_id.string: %s, count: %d\n",
+				__func__, info->hardware_id.string, cid_list->count);
```

Comment:
printing `info->hardware_id.string` might not be a great idea? ACPI_VALID_HID is not set so info->hardware_id might be not valid? Let's hope it is NULL or empty string and not going to lead to a kernel crash....

Reply:
@ujfalusi you could try other things such as 
u32 name;
which I believe is the 4-character code in the Device() definition, e.g. 

Device (CHUB)
Device (ESSX)

That would point directly to the definition in the DSDT?

https://elixir.bootlin.com/linux/latest/source/include/acpi/actypes.h#L1164